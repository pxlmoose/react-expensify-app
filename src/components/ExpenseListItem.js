import React from 'react';
import { Link } from 'react-router-dom'


const ExpenseListItem = ({  id, description, amount, createdAt }) => (
    <div> 
        <Link to={`/edit/${id}`}>
            <h3>{description}</h3>
        </Link>
        <p>{amount} - {createdAt}</p>
    </div>  
);


export default ExpenseListItem;

// 1. Import action you want to use 
// 2. You destructured props to use in html
// 3 On click you use imported action with given prop