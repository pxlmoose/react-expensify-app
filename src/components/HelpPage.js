import React from 'react';

const HelpPage = () => (
    <div>
    this is help component
    </div>
);

export default HelpPage;