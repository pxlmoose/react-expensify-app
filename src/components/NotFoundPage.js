import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
     {/* Link will make sure to use client side routing to pages you control (so no twitter links)*/}
    404! - <Link to="/">Go home</Link>         
    </div>
);

export default NotFoundPage;