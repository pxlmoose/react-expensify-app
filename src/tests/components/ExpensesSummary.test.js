import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render Expenses sumary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render Expenses sumary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={12436524} />);
    expect(wrapper).toMatchSnapshot();
});