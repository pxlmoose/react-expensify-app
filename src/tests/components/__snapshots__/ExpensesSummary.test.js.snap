// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Expenses sumary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render Expenses sumary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $124,365.24
  </h1>
</div>
`;
